shttps/Connection.cpp

```diff
@@ -471,21 +470,24 @@ namespace shttps {
             }
             else if (method_in == "POST") {
                 _method = POST;
-                if (content_length > 0) {
-                    vector <string> content_type_opts = process_header_value(header_in["content-type"]);
-                    if (content_type_opts[0] == "application/x-www-form-urlencoded") {
-                        char *bodybuf = nullptr;
-                        if (_chunked_transfer_in) {
-                            char *chunk_buf;
-                            ChunkReader ckrd(ins);
-                            content_length = ckrd.readAll(&chunk_buf);
-                            if ((bodybuf = (char *) malloc((content_length + 1) * sizeof(char))) == nullptr) {
-                                throw Error(__file__, __LINE__, "malloc failed!", errno);
-                            }
-                            memcpy(_content, chunk_buf, content_length);
-                            free(chunk_buf);
+                vector <string> content_type_opts = process_header_value(header_in["content-type"]);
+                if (content_type_opts[0] == "application/x-www-form-urlencoded") {
+                    char *bodybuf = nullptr;
+                    if (_chunked_transfer_in) {
+                        char *chunk_buf;
+                        ChunkReader ckrd(ins, _server->max_post_size());
+                        content_length = ckrd.readAll(&chunk_buf);
+                        if ((bodybuf = (char *) malloc((content_length + 1) * sizeof(char))) == nullptr) {
+                            throw Error(__file__, __LINE__, "malloc failed!", errno);
                         }
-                        else {
+                        memcpy(bodybuf, chunk_buf, content_length);
+                        free(chunk_buf);
```

Comment:
Right with exceptions. Since I'm using malloc and realloc which is more efficient, since it just increases the heap block if possible and copies the data only to the end of the heap if the block is too small. Thus it helps against fragmentation of the heap and is often much faster. String would always internally copy to a new heap location.